frontend/model/state.js

```diff
@@ -499,8 +579,8 @@ const getters = {
   },
   currentWelcomeInvite (state, getters) {
     const invites = getters.currentGroupState.invites
-    const inviteId = Object.keys(invites).find(invite => invites[invite].creatorID === INVITE_INITIAL_CREATOR)
-    const expires = getters.currentGroupState._vm.authorizedKeys[inviteId].meta.expires
+    const inviteId = Object.keys(invites).find(invite => invites[invite].creatorID === INVITE_INITIAL_CREATOR && !(getters.currentGroupState._vm.invites[invite].expires >= Date.now()) && !(getters.currentGroupState._vm.invites[invite].quantity <= 0) && getters.currentGroupState._vm.invites[invite].status === INVITE_STATUS.VALID)
```

Comment:
When lines of code grow beyond a certain point with logic that starts to break mere mortals brains, can you please, instead of inlining it like this, create a utility function that is called instead?

It will help improve code readability and potentially get rid of DRY violations too.